src/Test/Utilities/Portable/CommonTestBase.cs

```diff
@@ -189,17 +189,22 @@ static internal void RunValidators(CompilationVerifier verifier, Action<PEAssemb
         {
             if (assemblyValidator != null)
             {
-                using (var emittedMetadata = AssemblyMetadata.Create(verifier.GetAllModuleMetadata()))
-                {
-                    assemblyValidator(emittedMetadata.GetAssembly());
-                }
+                var emittedMetadata = verifier.GetMetadata();
+                Assert.Equal(MetadataImageKind.Assembly, emittedMetadata.Kind);
+
+                var assembly = ((AssemblyMetadata)emittedMetadata).GetAssembly();
+                assemblyValidator(assembly);
             }
 
             if (symbolValidator != null)
             {
-                var peModuleSymbol = verifier.GetModuleSymbolForEmittedImage();
-                Debug.Assert(peModuleSymbol != null);
-                symbolValidator(peModuleSymbol);
+                var emittedMetadata = verifier.GetMetadata();
```

Comment:
>verifier.GetMetadata() [](start = 38, length = 22)

Would it make sense to reuse result of calling this method across both if statements?